snuba_sdk/query_optimizers/or_optimizer.py

```diff
@@ -0,0 +1,68 @@
+from snuba_sdk.conditions import (
+    BooleanCondition,
+    BooleanOp,
+    Condition,
+    Op,
+)
+from snuba_sdk.formula import Formula
+from snuba_sdk.timeseries import Timeseries
+
+
+class OrOptimizer:
+    def optimize(self, query: Formula | Timeseries) -> Formula | Timeseries:
+        if query.filters is None:
+            return query
+
+        optimized = False
+        new_filters = []
+        for cond in query.filters:
+            res = self._optimize_condition(cond)
+            if res is not None:
+                optimized = True
+                new_filters.append(res)
+            else:
+                new_filters.append(cond)
+
+        if not optimized:
+            return query
+        elif isinstance(query, Timeseries):
+            return Timeseries(
```

Comment:
This can be simplified a bit, the `dataclass` library provides the `replace` function:

```
replace(query, filters=new_filters)
```

This is equivalent to what you're doing here.

Reply:
Thank Evan, I think I need to read up on dataclasses a bit